### ISSUE TITLE
Do you have display problem in Linux?

### ISSUE THREAD

**Issue by nandttl** [boutch55555/x2x0-agan#1]
Excuse me, I installed the agan's x230 fhd mod too, but my screen is often black after grub, and I have to use nomodeset kernel command line to get the screen work. But Ubuntu gets really slow after using nomodset. And I have tried a lot of method, all doesn't work. By the way, the screen works fine in Windows. I've tried on Ubuntu 18.04, debian stretch and debian buster, all the same problem. Thank you

**Comment by boutch55555** [comment]
I used to have a similar problem on Debian Stretch. Can't recall if it was on the same motherboard I'm currently using. I also had this issue with nitrocaster's board. Really odd, I would only get the panel to work 1 in about 10 boots, regardless what was set as default display in the bios. Test waiting until there is low HDD activity, indicating you're at the login screen. Press power 2 sec to put it to sleep. The display should show up when you wake it back. My current build doesn't have this bug, no idea why.

**Comment by nandttl** [comment]
Yes! The same method works for me too! After entering the login screen, sleep it and wake up again, the screen works. But if I leave the computer for a while and the screen is off, the screen won't work and need another sleep to make it work again. And I found in ubuntu, xrandr will report "the display is not connected but have no modes". I'm sure the EDID information can be read by the read-edid tools. I will keep working on it. Thank you for your share!

**Comment by nandttl** [comment]
And by the way, what's your current build? I'd like to try it. Thank you!

**Comment by ghost** [comment]
Hi, I'm having the same issue. Are y'all running coreboot or stock bios?

**Comment by nandttl** [comment]
> Hi, I'm having the same issue. Are y'all running coreboot or stock bios?

I'm using stock bios but modified the white list.

**Comment by ghost** [comment]
I was running the original bios when I saw this issue, but now I'm running coreboot and seeing the same issue as soon as Linux boots.

I guess what's going on is the display mode can not be set correctly sometimes, and it switches to use the LVDS output only. With the following patch it retries setting the display mode until it works:

```
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
--- a/drivers/gpu/drm/i915/intel_dp.c	2019-06-04 07:59:45.000000000 +0200
+++ b/drivers/gpu/drm/i915/intel_dp.c	2019-06-08 23:39:14.560262180 +0200
@@ -124,7 +124,7 @@
 {
 	struct intel_digital_port *intel_dig_port = dp_to_dig_port(intel_dp);

-	return intel_dig_port->base.type == INTEL_OUTPUT_EDP;
+	return intel_dig_port->base.type == INTEL_OUTPUT_EDP || strncmp(intel_dp->attached_connector->base.name, "DP-3",4) == 0;
 }

 static struct intel_dp *intel_attached_dp(struct drm_connector *connector)
@@ -5977,7 +5977,7 @@
 	if (INTEL_GEN(dev_priv) < 5)
 		return false;

-	if (INTEL_GEN(dev_priv) < 9 && port == PORT_A)
+	if (INTEL_GEN(dev_priv) < 9 && port == PORT_D)
 		return true;

 	return intel_bios_is_port_edp(dev_priv, port);
diff --git a/drivers/gpu/drm/i915/intel_lvds.c b/drivers/gpu/drm/i915/intel_lvds.c
--- a/drivers/gpu/drm/i915/intel_lvds.c	2019-06-04 07:59:45.000000000 +0200
+++ b/drivers/gpu/drm/i915/intel_lvds.c	2019-06-08 23:40:36.153071061 +0200
@@ -502,6 +502,13 @@
 /* These systems claim to have LVDS, but really don't */
 static const struct dmi_system_id intel_no_lvds[] = {
 	{
+		.ident = "Modded ThinkPad X230",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "2330A17"),
+		},
+	},
+	{
 		.callback = intel_no_lvds_dmi_callback,
 		.ident = "Apple Mac Mini (Core series)",
 		.matches = {
```

**Comment by ghost** [comment]
I talked to @lynxis about this issue, who suggested using coreboot with this patch: https://review.coreboot.org/c/coreboot/+/28950
After testing it I can say it has the same effect as the kernel patch above. Display init fails a random number of times (the backlight flashes on and off) until it succeeds. Since LVDS is disabled with this patch it does not give up and switch to LVDS output.

**Comment by nandttl** [comment]
May be the problem is in kms? As the screen works fine after using --nomodeset.


|
Nand Shen
邮箱：nandttl@126.com
|
签名由网易邮箱大师定制
On 8/9/2019 16:30，Milan<notifications@github.com> wrote：

I talked to @lynxis about this issue, who suggested using coreboot with this patch: https://review.coreboot.org/c/coreboot/+/28950
After testing it I can say it has the same effect as the kernel patch above. Display init fails a random number of times (the backlight flashes on and off) until it succeeds. Since LVDS is disabled with this patch it does not give up and switch to LVDS output.

**Comment by drewth1** [comment]
Sorry what’s the best way to fix Linux during this boot issue we’ve been having?

**Comment by nandttl** [comment]
> Sorry what’s the best way to fix Linux during this boot issue we’ve been having?

disable kms with nomodeset in kernel command line. that's what i know. maybe you will suffer some very odd display experience, like extreme low refresh rate.

**Comment by drewth1** [comment]
Yep am experiencing very low refresh rate.
